indico/modules/categories/controllers/display.py

```diff
@@ -625,24 +635,38 @@ def _get_event_data(self, event_query):
                 rooms[room.id] = {'id': room.id,
                                   'title': room.full_name,
                                   'venueId': room.location_id}
-            if self.group_by == self.GroupBy.category:
-                comparison_id = category_id
-            elif self.group_by == self.GroupBy.location:
-                comparison_id = event.own_venue_id or 0
-            else:
-                comparison_id = room.id if room else 0
             event_data = {'title': event.title,
                           'start': event.start_dt.astimezone(tz).replace(tzinfo=None).isoformat(),
                           'end': event.end_dt.astimezone(tz).replace(tzinfo=None).isoformat(),
                           'url': event.url,
                           'categoryId': category_id,
                           'venueId': event.own_venue_id,
+                          'keywords': event.keywords,
                           'roomId': room.id if room else None}
-            colors = generate_contrast_colors(comparison_id)
+            if self.group_by == self.GroupBy.category:
+                colors = generate_contrast_colors(category_id)
+            elif self.group_by == self.GroupBy.location:
+                colors = generate_contrast_colors(event.own_venue_id or 0)
+            elif self.group_by == self.GroupBy.room:
+                colors = generate_contrast_colors(room.id if room else 0)
+            else:
+                # by keywords
+                if any(kw not in allowed_keywords for kw in event.keywords):
+                    keyword_id = -1
+                elif len(event.keywords) == 0:
```

Comment:
```suggestion
                elif not event.keywords:
```

(but maybe make this the first `if` case?)

Reply:
Changed!